src/frontend/app/components/modals/TestEditModal/TestEditModal.test.tsx

```diff
@@ -1,172 +1,197 @@
-    },
-  };
-  const mockStore = mockedStore(defaultStore);
   const defaultProps = {
     onClose: jest.fn(),
     open: true,
     showClose: true,
   };
 
-  it("renders an empty form", async () => {
-    render(<TestEditModal {...defaultProps} />, mockStore);
-
-    screen.getByText("New Test");
-  });
-
-  it("renders the languages", () => {
-    render(<TestEditModal {...defaultProps} />, mockStore);
+  const languages = ["en-US", "es-ES", "fr-FR"];
 
-    const selectElement = screen.getByPlaceholderText("Select the language");
-    fireEvent.click(selectElement);
-    languages.forEach((lang) => {
-      screen.getAllByText(`${lang.attributes.longName} - ${lang.attributes.shortRegion}`);
+  beforeEach(() => {
+    jest.clearAllMocks();
+    
+    // Default mock setup
+    mockUseGetLanguagesQuery.mockReturnValue({
+      data: languages,
+      isLoading: false,
     });
+    mockUseGetAllTestsQuery.mockReturnValue({
+      data: [],
+      isLoading: false,
+    });
+    mockUseCreateNewTestMutation.mockReturnValue([mockCreateTest, { isLoading: false }]);
+    mockUseEditTestMutation.mockReturnValue([mockUpdateTest, { isLoading: false }]);
   });
 
-  describe("with empty store", () => {
-    it("dispatches getCertifications call", () => {
-      const emptyCerts = { ...defaultStore, certifications: { certifications: [] } };
-      const emptyCertStore = mockedStore(emptyCerts);
-
-      render(<TestEditModal {...defaultProps} />, emptyCertStore);
-
-      expect(emptyCertStore.getActions()).toEqual([
-        { payload: undefined, type: "certifications/getCertificationsStart" },
-      ]);
+  describe("New Test Mode", () => {
+    it("renders an empty form for new test with essential elements", () => {
+      render(<TestEditModal {...defaultProps} />);
+
+      expect(screen.getByText("New Test")).toBeInTheDocument();
+      expect(screen.getByText("Title")).toBeInTheDocument();
+      expect(screen.getByText("Description")).toBeInTheDocument();
+      expect(screen.getByText("Language")).toBeInTheDocument();
+      expect(screen.getByText("Level")).toBeInTheDocument();
+      expect(screen.getByText("Version")).toBeInTheDocument();
+      expect(screen.getByText("Done")).toBeInTheDocument();
     });
 
-    it("dispatches getLanguages call", () => {
-      const emptyLangs = { ...defaultStore, languages: { languages: [] } };
-      const emptyLangStore = mockedStore(emptyLangs);
+    it("handles input changes for title field", async () => {
+      const user = userEvent.setup();
+      render(<TestEditModal {...defaultProps} />);
 
-      render(<TestEditModal {...defaultProps} />, emptyLangStore);
+      const titleInput = screen.getByLabelText("title");
+      await user.type(titleInput, "New Test Title");
 
-      expect(emptyLangStore.getActions()).toEqual([
-        {
-          payload: undefined,
-          type: "languages/getLanguagesStart",
-        },
-      ]);
+      expect(titleInput).toHaveValue("New Test Title");
```

Comment:
Would these input checking tests make more sense wrt to checking not the input control but the mutation being called once all mandatory fields are filled in? Think about it - don't take my suggestion at face value. What are pros and cons of approaches here?